# CryptoProphet
## Notebook's Goal
> Check extracted tweets for the analysis (amount and influencers)

In [1]:
# imports packages
from src.helpers import load_df_tweets
import numpy as np

# loads data
df = load_df_tweets()

# prints shape
print(df.shape)
print(f"Tweets from {df['created_at'].min()} to {df['created_at'].max()}")

# groups by influencer
df\
    .groupby(['user_screen_name'])\
    .agg({'created_at': [np.min,np.max,len]})\
    .sort_values([('created_at','amax')], ascending=False)

100%|███████████████████████████████████████████████████████████████████████| 127642/127642 [00:01<00:00, 86938.00it/s]


(127642, 36)
Tweets from 2008-09-11 15:35:45+00:00 to 2021-10-13 19:51:34+00:00


created_at                                
                                      amin                      amax   len
user_screen_name                                                          
AlertTrade       2021-10-08 13:10:38+00:00 2021-10-13 19:51:34+00:00  3250
AltcoinDailyio   2021-05-20 20:02:41+00:00 2021-10-13 19:43:14+00:00  3246
intocryptoverse  2021-05-06 23:09:51+00:00 2021-10-13 19:40:54+00:00  3250
TheCryptoLark    2021-07-01 03:18:53+00:00 2021-10-13 19:31:42+00:00  3250
whale_alert      2021-08-31 15:17:54+00:00 2021-10-13 19:30:21+00:00  3250
coinbureau       2020-07-08 15:04:02+00:00 2021-10-13 19:20:14+00:00  3250
TheMoonCarl      2021-03-08 16:42:13+00:00 2021-10-13 18:30:07+00:00  3250
Cardano          2018-09-24 18:39:40+00:00 2021-10-13 15:42:16+00:00  3205
StackerSatoshi   2020-06-04 17:27:30+00:00 2021-10-13 15:39:52+00:00  1767
100trillionUSD   2020-08-20 09:14:29+00:00 2021-10-13 14:00:11+00:00  3249
ethereum         2014-01-29 19:00:51+00:00 2021-10-12 18:07:32+00:00  3124
Maxwell_Maher    2012-10-02 15:12:41+00:00 2021-10-11 20:57:04+00:00   128
BTCFoundation    2014-04-29 12:45:42+00:00 2021-10-11 12:29:15+00:00  3180
Cryptosrus       2018-01-06 16:47:51+00:00 2021-10-11 07:44:05+00:00   176
APompliano       2021-02-14 00:53:11+00:00 2021-06-11 00:24:57+00:00  3449
CryptoWendyO     2021-03-06 21:57:00+00:00 2021-06-11 00:22:25+00:00  3436
mysticaloaks     2021-03-11 03:22:37+00:00 2021-06-11 00:22:09+00:00  3450
TyDanielSmith    2018-03-13 01:34:41+00:00 2021-06-10 23:57:39+00:00  2938
davidgokhshtein  2021-04-03 03:21:59+00:00 2021-06-10 23:46:03+00:00  3450
CryptoCobain     2021-03-16 06:33:21+00:00 2021-06-10 23:40:28+00:00  3445
KennethBosak     2021-03-30 23:57:20+00:00 2021-06-10 22:58:26+00:00  3450
girlgone_crypto  2020-11-30 15:40:54+00:00 2021-06-10 21:38:50+00:00  3448
elonmusk         2020-06-11 06:35:23+00:00 2021-06-10 21:27:17+00:00  3249
BarrySilbert     2017-07-27 03:02:34+00:00 2021-06-10 20:34:50+00:00  3428
ToneVays         2020-06-18 03:28:01+00:00 2021-06-10 19:02:50+00:00  3446
IvanOnTech       2020-06-15 10:49:11+00:00 2021-06-10 18:38:48+00:00  3416
ErikVoorhees     2020-09-20 18:20:32+00:00 2021-06-10 16:55:17+00:00  3449
officialmcafee   2020-08-29 20:47:35+00:00 2021-06-10 16:00:14+00:00  3241
LayahHeilpern    2021-05-14 07:15:14+00:00 2021-06-10 15:12:51+00:00  3246
danheld          2021-03-31 00:35:29+00:00 2021-06-10 14:57:26+00:00  3248
TheCryptoDog     2021-04-02 08:58:52+00:00 2021-06-10 14:52:32+00:00  3249
Bitboy_Crypto    2020-07-30 11:31:50+00:00 2021-06-10 14:41:04+00:00  3243
aantonop         2020-09-01 22:27:19+00:00 2021-06-10 13:51:43+00:00  3321
justinsuntron    2020-06-26 08:46:56+00:00 2021-06-10 13:46:03+00:00  3412
PeterMcCormack   2021-04-27 04:21:12+00:00 2021-06-10 04:38:21+00:00  3249
bhorowitz        2008-09-11 15:35:45+00:00 2021-06-10 04:17:10+00:00   980
brian_armstrong  2020-11-01 01:52:15+00:00 2021-06-10 02:27:51+00:00   335
Coinbound_io     2019-10-12 01:50:51+00:00 2021-06-10 01:13:16+00:00   882
SatoshiLite      2018-07-11 20:46:35+00:00 2021-06-09 17:44:01+00:00  3258
CryptoEuclid     2021-03-05 18:26:13+00:00 2021-06-08 19:01:39+00:00  3341
EuclidAndOaks    2018-01-09 01:41:13+00:00 2021-06-08 15:56:08+00:00   848
TimDraper        2009-02-15 01:09:39+00:00 2021-06-08 04:49:57+00:00  2307
VitalikButerin   2020-06-12 21:16:46+00:00 2021-06-07 22:53:32+00:00  3204
FEhrsam          2011-06-12 19:35:24+00:00 2021-06-01 22:16:24+00:00  1440
StackingUSD      2021-02-17 19:46:58+00:00 2021-05-04 00:17:37+00:00     3
rogerkver        2013-04-01 06:24:53+00:00 2021-04-26 16:04:55+00:00  3203
CryptoBrekkie    2019-11-25 02:27:37+00:00 2020-07-18 16:37:35+00:00     3

# Conclusion
> Tweets properly extracted.

### ⚠️ Limitation
> Twitter API free tier only enables extraction of 3250 tweets from each user's timeline.